### PR TITLE
Inline a few trivial private methods

### PR DESCRIPTION
**Pull request by thiemowmde**
This is a direct follow up for #205.

[Bug: T157959](https://phabricator.wikimedia.org/T157959)

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
This gains us what? (I think inlining requires justification with numbers)

**Comment by thiemowmde** [comment]
Another 1% gain in runtime, faster and easier to read code (e.g. native `is_string` instead of assertIsString).

File: src/Deserializers/StatementDeserializer.php [hunk 1]
```diff
@@ -61,15 +61,9 @@ public function __construct(
...
-
-	private function hasType( $serialization ) {
-		return is_array( $serialization ) && array_key_exists( 'type', $serialization );
-	}
-
-	private function hasCorrectType( $serialization ) {
-		return in_array( $serialization['type'], array( 'claim', 'statement' ) );
+		return is_array( $serialization )
+			&& array_key_exists( 'type', $serialization )
+			&& ( $serialization['type'] === 'claim' || $serialization['type'] === 'statement' );
```
Comment (bekh6ex) [inline src/Deserializers/StatementDeserializer.php]: IMHO, It was more clear before.

Comment (thiemowmde) [inline src/Deserializers/StatementDeserializer.php]: I disagree. What puzzled me the most was the code that checked if a variable "has type" and "has correct type". What "type"? Does this check if the variable is an array? How can such a type check be "correct"?

If you find `in_array( $serialization['type'], [ 'claim', 'statement' ] )` more readable we can change the last line.

Comment (JeroenDeDauw) [inline src/Deserializers/StatementDeserializer.php]: The method was badly named. `hasTypeKey` would be better.

**Comment by JeroenDeDauw** [comment]
I think it's a bad idea to inline code to get a 1% runtime gain, which simply is not noticeable once you factor in i/o for nearly all usecases. For performance issues I'd also start by looking at what actually annoys the users. The speed of a maintenance script does likely not rate high there. And much bigger gains might be made in other ways (such as not having ones code run on shitty old PHP versions).

**Comment by manicki** [comment]
+1
I am not convinced by the impact of this but I not opposing.
I'd be also fine with renaming that private method instead.

File: src/Deserializers/StatementDeserializer.php (hunk 1 above)
Comment (thiemowmde) [inline src/Deserializers/StatementDeserializer.php]: I reworked this method into one that is called `isValidClaimType` now.

**Comment by manicki** [comment]
Still looks fine to me after the last change.

**Pull request merged**